### DISCUSSION TITLE
Question on Pod Reuse

### DISCUSSION THREAD

**Discussion by cuserdzy** [argoproj/argo-workflows#12255] [Q&A]
For this question, I have searched through issues/discussions and found that Argo definitely doesn't support pod reuse, for example https://github.com/argoproj/argo-workflows/discussions/5423.

I understand this might be due to reasons such as env isolation, since the task in each step is not a service. However, in most scenarios, I have multiple workflows that might share same tasks and my task is stateless. Therefore, I find it unnecessary to boot up a new container each time my workflow executes this step. I wonder, for a frequently called and stateless task, could the container possibly be reused?

I think the scenarios might be similar to using Step Function with Lambda, although I'm not very familiar with this. I guess that Lambda won't always cold boot when called by Step Function.

So currently, I'm considering the possibility of using OpenFaas with Argo, where OpenFaas handles the function lifecycle and Argo is used to schedule these serverless functions. However, the feasibility of this direction is not very clear to me, If anyone has had a similar idea or already has some experience with this, I would be very happy to learn.

**Comment by jrbe228** [comment]
WASM execution via this ArgoWF [plugin ](https://github.com/Shark/wasm-workflows-plugin)could be one approach for serverless.

**Reply by tooptoop4** [reply]
deadlink?

**Reply by jrbe228** [reply]
Thanks, the link should be fixed now.
